### PR TITLE
fix: create spot instance service-linked role for Karpenter

### PR DESCRIPTION
**Pull request by sgametrio**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -184,12 +184,15 @@ resource "aws_route_table_association" "karpenter" {
...
 
+  version = "20.8.5"
+
   cluster_name                    = module.eks.cluster_name
   enable_irsa                     = true
   irsa_oidc_provider_arn          = module.eks.oidc_provider_arn
   irsa_namespace_service_accounts = ["kube-system:karpenter"]
   iam_role_name                   = "karpenter-${local.id}"
   iam_role_use_name_prefix        = false
+  create_instance_profile         = true
```
Comment (swibrow) [inline main.tf:195]: Karpenter creates instance profiles itself so this should be disabled by default

File: main.tf [hunk 2]
```diff
@@ -201,6 +204,15 @@ module "karpenter" {
   tags = local.tags
 }
 
+# Allow creation of SPOT instances
+resource "aws_iam_service_linked_role" "spot" {
```
Comment (swibrow) [inline main.tf:208]: Can only have one per account so should not be in this module.

File: main.tf [hunk 3]
```diff
@@ -201,6 +204,15 @@ module "karpenter" {
...
+  aws_service_name = "spot.amazonaws.com"
+}
+
+resource "aws_iam_service_linked_role" "spot_fleet" {
```
Comment (swibrow) [inline main.tf:212]: Same here

**Review by swibrow — CHANGES_REQUESTED** [review]
Create the linked roles out of the module. I had the idea to create a bootstrap module that managed the required settings per account.

I guess you tried to deploy to a new account so you had issues with the instance profile, but it's not required once the service linked roles are created.

**Comment by sgametrio** [comment]
@swibrow Thanks for the feedback. Yes I tried running this in the ATM Staging account. I will update the PR to explicit the dependencies that this module needs in order to run with no issues.

I'm good with an external bootstrap module.

**Comment by sgametrio** [comment]
Closed in favor of #15

**Pull request closed without merging**